dashboard/signals.py

```diff
@@ -0,0 +1,30 @@
+"""
+Signals for user profiles
+"""
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from dashboard.models import CachedEnrollment, ProgramEnrollment
+
+
+@receiver(post_save, sender=CachedEnrollment, dispatch_uid="update_programenrollment")
+def create_programenrollment(sender, instance, created, **kwargs):  # pylint: disable=unused-argument
+    """
+    Signal handler to create Program enrollment when the CachedEnrollment table is updated
+    """
+    if instance.data is not None:
+        ProgramEnrollment.objects.get_or_create(
+            user=instance.user,
+            program=instance.course_run.course.program
+        )
+    else:
+        # check if there is any other entry with data != None
+        # if not delete the enrollment
+        if CachedEnrollment.objects.filter(
+            user=instance.user,
+            course_run__course__program=instance.course_run.course.program
+        ).exclude(data__isnull=True).count() == 0:
+            ProgramEnrollment.objects.filter(
+                user=instance.user,
+                program=instance.course_run.course.program
+            ).delete()
```

Comment:
about 2 and 3: ok

about 1: I understand your point but I am not convinced.
it is true that if I check for `created` I will limit the queries at insertion time, but the big amount of queries will be at "update" time, because it will be more likely that the entry for `CachedEnrollment` will be already in the DB (consider that there will be at minimum an update of the timestamp every 6 hours). So I do not see a significant increase of performance here.
Let me know if I am missing something